README.md

```diff
@@ -1,30 +1,63 @@
-<h1 align="center">OpenST - Empowering Decentralized Economies</h1>
+# 💠 Mosaic Contracts
 
-[![Gitter: JOIN CHAT](https://img.shields.io/badge/gitter-JOIN%20CHAT-brightgreen.svg)](https://gitter.im/OpenSTFoundation/SimpleToken)
+![Build master](https://img.shields.io/travis/OpenSTFoundation/mosaic-contracts/master.svg?label=build%20master&style=flat)
+![Build develop](https://img.shields.io/travis/OpenSTFoundation/mosaic-contracts/develop.svg?label=build%20develop&style=flat)
+![npm version](https://img.shields.io/npm/v/@openstfoundation/mosaic-contracts.svg?style=flat)
+[![Discuss on Discourse](https://img.shields.io/discourse/https/discuss.openst.org/topics.svg?style=flat)](https://discuss.openst.org/)
+[![Chat on Gitter](https://img.shields.io/gitter/room/OpenSTFoundation/SimpleToken.svg?style=flat)](https://gitter.im/OpenSTFoundation/SimpleToken)
 
-[OpenST](https://openst.org/) blockchain infrastructure empowers decentralized economies. The central component of this infrastructure is the OpenST Protocol, a framework for building scalable blockchain token economies.
+Mosaic is a parallelization schema for decentralized applications.
+It composes heterogeneous blockchain systems into one another.
+Decentralized applications can use Mosaic to compute over a composed network of multiple blockchain systems in parallel.
 
-_While OpenST is available as-is for anyone to use, we caution that this is early stage software and under heavy ongoing development and improvement. Please report bugs and suggested improvements._
+Mosaic enables building scalable blockchain token economies through the bidirectional transposition of ERC20 tokens on one blockchain, the *origin* chain, and a utility token representation on another blockchain, the *auxiliary* chain.
 
-## OpenST Protocol
-The OpenST Protocol enables building scalable blockchain token economies through the bidirectional transposition of value tokens on one blockchain, the "value chain", and a utility token representation on another blockchain, the "utility chain".
+The Protocol defines a set of actions that are performed atomically across a gateway. A gateway for a given token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 contract on auxiliary that mints and burns utility tokens for an equivalent value of ERC20 tokens staked and unstaked on origin. Atomicity is achieved by combining performance of the transpositional actions in a 2-phase-commit structure with a hash-timelock on each of the origin and auxiliary chains.
```

Comment:
The protocol defines a set of actions that together perform atomic token transfers on two blockchains across a gateway. A gateway for a given EIP20 token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 utility token contract on auxiliary that mints and burns utility tokens to atomically mirror tokens staked and unstaked on the originchain.

Atomicity is achieved using a 2-phase message passing architecture between the chains. Messages are declared on the source chain, and confirmed on the target chain once the source chain is finalized with Patricia Merkle proofs.  Once messages are confirmed on the target chain, they can efficiently progressed with a hashlock.
Messages can also be reverted if they are not yet completed on the target chain.

Reply:
🤩